src/main/scala/utils/Connections.scala

```diff
@@ -205,40 +210,44 @@ object Connections {
     * @return
     *   a list of unique [[Connection]]s
     */
-  def buildUniqueConnections[T](
+  def buildUndirectedShortestPathConnections[T](
       graph: Graph[T, DistanceWeightedEdge],
       shortestPath: ShortestPathAlgorithm[T, DistanceWeightedEdge]
   ): List[Connection[T]] = {
     val vertexes = graph.vertexSet().asScala
-    val paths = vertexes.map { v => shortestPath.getPaths(v) }.toList
-
-    paths.flatMap { p =>
-      vertexes.flatMap { v =>
-        if (p.getSourceVertex != v) {
-          val path = p.getPath(v)
-          Some(
-            Connection(
-              p.getSourceVertex,
-              v,
-              Quantities.getQuantity(path.getWeight, Units.METRE),
-              Some(path)
-            )
+    val paths = vertexes.map { v => v -> shortestPath.getPaths(v) }.toMap
+
+    getAllUniqueCombinations(graph.vertexSet().asScala.toList).map {
+      case (nodeA, nodeB) =>
+        val path = paths(nodeA).getPath(nodeB)
+
+        if (path == null) {
+          val unconnected =
+            if (graph.edgesOf(nodeA).size() == 0) nodeA else nodeB
```

Comment:
I don't think we can say that - the unconnected node could still have edges to other nodes that are also unconnected to the other component of the graph. So I'd keep the error message more general, e.g.
"No path could be found between $nodeA and $nodeB, because either node is not connected to the graph."